build_tools/upload_logs_to_s3.py

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env python3
+import os
+import glob
+import subprocess
+
+
+def run_aws_cp(src, dest, content_type=None):
+    cmd = (
+        ["aws", "s3", "cp", src, dest, "--recursive"]
+        if os.path.isdir(src)
+        else ["aws", "s3", "cp", src, dest]
+    )
+    if content_type:
+        cmd += ["--content-type", content_type]
+    try:
+        print(f"[INFO] Running: {' '.join(cmd)}")
+        subprocess.run(cmd, check=True)
+    except subprocess.CalledProcessError as e:
+        print(f"[ERROR] Failed to upload {src} to {dest}: {e}")
+
+
+def upload_logs_to_s3(bucket_name, run_id, amdgpu_family, log_dir="build/logs"):
+    s3_base_path = f"s3://{bucket_name}/{run_id}/logs/{amdgpu_family}"
+
+    if not os.path.isdir(log_dir):
+        print("[INFO] No logs directory found. Creating it.")
```

Comment:
Needs to be updated to reflect the latest changes. Furthermore, instead of using `print` you could consider introducing

```python
def log(*args, **kwargs):
    print(*args, **kwargs)
    sys.stdout.flush()
```

and use that like we do in other Python scripts.

Reply:
using log now